instructions.json

```diff
@@ -0,0 +1,19 @@
+{"instructions": 
+  {
+    "db_name":"apidb",
+	"user_name" : "user",
+	"password" : "password",
+	"region_name": "New Zealand",
+	
+	"url": "https://data.linz.govt.nz/layer/101292-nz-building-outlines-all-sources/",
+	"api" : "https://data.linz.govt.nz/services;key=API_KEY/wfs?service=WFS&version=2.0.0&request=GetFeature&typeNames=layer-101292&outputFormat=json&SRSName=EPSG:2193&cql_filter=bbox(shape,5169354.082, 1559525.958, 5167380.381, 1558247.433 )",
+	"apis": {
+		"linz": {
+				"source_name":"Building_Outlines",
+				"geometry_col_name": "shape"
+		}
+	},
+	"sql_query":"CREATE TABLE IF NOT EXISTS public.apilinks(unique_id uuid NOT NULL DEFAULT uuid_generate_v4(),source_name character varying(255),source_apis character varying(65535),access_date timestamp without time zone DEFAULT CURRENT_TIMESTAMP,url character varying(67777),api_modified_date date,region_name character varying(255),geometry geometry,geomid character varying DEFAULT 'nz_1'::character varying,api_key character varying(255),query_dictionary jsonb,geometry_col_name character varying(255),UNIQUE (source_name))",
+	"insert_query":"INSERT INTO apilinks (source_name,source_apis, url,region_name,query_dictionary,api_modified_date,geometry_col_name) VALUES (%s,%s,%s,%s,%s,%s,%s);UPDATE apilinks SET geometry =(SELECT geom FROM nz_polygons WHERE apilinks.geomid = nz_polygons.geomid)"
```

Comment:
It would be good to use SQL prepared statements or parameterized queries in Python. A post about it can be found on [stackoverflow here](https://stackoverflow.com/questions/19314342/python-sqlalchemy-pass-parameters-in-connection-execute). This would reduce the chance of injection attacks, then you only have the variables that are going to change in the query in the instructions.json. Another way would be having a series of `.sql` files that are scripts to set up the database or perform the more frequent queries but you can do this in python too